src/App.tsx

```diff
@@ -298,18 +327,35 @@ const App = () => {
     isAuthed,
     characterData,
     isFetchingCharacterData,
-    isBungieSystemDisabled
+    isBungieSystemDisabled,
   ]);
 
   const onSelectMembership = useCallback((membership: UserInfoCard) => {
     ga.event({
       category: "Platform",
       action: "Select platform",
-      label: `Membership type: ${membership.membershipType}`
+      label: `Membership type: ${membership.membershipType}`,
     });
     setSelectedDestinyMembership(membership);
   }, []);
 
+  const onRefreshClick = () => {
+    setIsFetchingCharacterData(true);
+
+    try {
+      doHardRefresh(
+        setCharacterData,
+        setBungieSystemDisabled,
+        setBungieServiceUnavailable,
+        characterData
+      );
+    } catch (e) {
+      throw e;
+    } finally {
+      setIsFetchingCharacterData(false);
+    }
+  };
```

Comment:
the `setIsFetchingCharacterData(true);` calls here don't actually set loading to `true` and I couldnt figure out why, so the loading indicator does not display while the call to SetItemLockState takes place...

Reply:
actually I just set the value in the `bustProfileCache` in utils.